src/dao/DAO.js

```diff
@@ -190,26 +191,8 @@ class DAO<TEntity: { id: EntityID }, TEntityMutator> extends BaseDAO<
   }
 
   fetchCustom<TResult>(queryOptions?: QueryOptions): LoadObject<TResult> {
-    const cacheKey = this._getCacheKey(queryOptions);
-    if (!this._customLoaderByQuery.has(cacheKey)) {
-      this._customLoaderByQuery.set(cacheKey, LoadObject.loading());
-      this._emitChanges();
-
-      this.__resolve(this.__buildHandler(queryOptions))
-        .then((result: Object) => {
-          this._customLoaderByQuery.set(
-            cacheKey,
-            LoadObject.withValue(result.data),
-          );
-          this._emitChanges();
-        })
-        .catch((error: Error) => {
-          this._customLoaderByQuery.set(cacheKey, LoadObject.withValue(error));
-          this._emitChanges();
-        });
-    }
-
-    return nullthrows(this._customLoaderByQuery.get(cacheKey));
+    const cacheKey = this.__getCacheKey(queryOptions);
+    return this.__fetchCustom(this.__buildHandler(queryOptions), cacheKey);
```

Comment:
Why not just pass `queryOptions` into this.  It will keep the cache key generator private which is good. Nobody should have to care about the cache.